audit.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3.7
+
+import os
+import sys
+from subprocess import CalledProcessError, check_call
+import github3 as gh3
+
+DIR = os.path.abspath(os.path.dirname(__file__))
+WORK = os.path.join(DIR, "work")
+CARGO_HOME = os.path.join(DIR, ".cargo")
+RUSTUP_HOME = os.path.join(DIR, ".rustup")
+CARGO = os.path.join(CARGO_HOME, "bin", "cargo")
+SOFTDEV_GH = "https://github.com/softdevteam"
+RUSTUP_URL = "https://sh.rustup.rs/"
+GH_API_HOST = "api.github.com"
+GH_API_REPOS = "/users/softdevteam/repos"
+
+# If you want to skip any soft-dev repos, you can add them here.
+SD_SKIP_REPOS = []
+
+
+def get_sd_rust_repos(token_file):
+    """Get a list of unarchived soft-dev repos written in Rust"""
+
+    with open(token_file) as f:
+        token = f.read().strip()
+
+    gh = gh3.login(token=token)
+    return [r for r in gh.repositories() if
+            r.owner.login == "softdevteam" and
+            "Rust" in map(lambda tup: tup[0], r.languages()) and
+            not r.archived and
+            r.name not in SD_SKIP_REPOS]
+
+
+def install_cargo_audit():
+    os.environ["RUSTUP_HOME"] = RUSTUP_HOME
+    os.environ["CARGO_HOME"] = CARGO_HOME
+
+    check_call(["curl", "--proto", "=https", "--tlsv1.2", "-sSf",
+                "https://sh.rustup.rs", "-o", "rustup.sh"])
+    check_call(["sh", "rustup.sh", "--no-modify-path", "-y"])
+    check_call([CARGO, "install", "cargo-audit"])
+
+
+def audit(name, repo):
+    direc = os.path.join(WORK, name)
+
+    # Either pull or update the source from git.
+    src_exists = os.path.exists(direc)
+    if not src_exists:
+        os.chdir(WORK)
+        git_cmd = ["git", "clone", repo, name]
+    else:
+        os.chdir(direc)
+        git_cmd = ["git", "pull"]
+
+    try:
+        check_call(git_cmd)
+    except CalledProcessError:
+        return False
+
+    os.chdir(direc)
+
+    # If there's no Cargo.toml, we can't audit it.
+    if not os.path.exists("Cargo.toml"):
+        print("No Cargo.toml")
```

Comment:
Is it useful to print this? I think it'll just make the emails harder to read?

Reply:
These don't appear in the email, only in the buildbot log.

Not sure if that changes anything though. We can kill it if you want, but it just looked odd that it said it was going to audit the repo, then it doesn't ;)

Reply:
Ah, I see. OK, as you wish. I hate chatty scripts, but since I'm not the one responsible for monitoring this, I'm not going to worry about it :p

Reply:
I don't really mind the chattiness here since i will only every look at it if it fails. If it passes no-one ever sees the output since no email was sent.

(So it's like sysutils/supuner)

Sample log:
http://bencher13.soft-dev.org:8010/#/builders/4/builds/5/steps/3/logs/stdio